content/blog/2019/knative-activator-adapter/index.md

```diff
@@ -0,0 +1,93 @@
+an idle application receives a new request and needs to scale back from zero. Knative scale-to/from-zero is built on
+a number of internal components with a precise and complex interaction.
+
+The diagram below presents the current Knative architecture for scale-from-zero.
+
+{{< image width="60%" link="knative-activator.png" caption="Knative scale-from-zero" >}}
+
+The traffic for an idle application is redirected to **Activator** component by programming Istio with *VirtualServices*
+and *DestinationRules*. When **Activator** receives a new request, it:
+
+1. buffers incoming requests
+
+1. triggers to **Autoscaler**
+
+1. redirects requests to the application after it has been scaled up, including retries and load-balancing (if needed)
+
+Once the application is up and running again, Knative restores the routing from **Activator** to the running application.
+
+## Mixer adapter
+
+[Mixer](https://istio.io/faq/mixer/) provides a rich intermediation layer between the Istio components and infrastructure backends.
+It is designed as a stand-alone component, separate from [Envoy](https://www.envoyproxy.io/), and has a simple extensibility model
+to allow Istio to interoperate with as widest breath of backends as possible. Due to its design, Mixer is inherently easier to extend
+than Envoy is.
+
+It is an attribute processing engine that uses operator-supplied configuration to map request attributes from the proxy into calls
+to these backend systems via a pluggable set of adapters. Adapters enable **Mixer** to expose a single consistent API, independent of the
+infrastructure backends in use. The exact set of adapters used at runtime is determined through operator configuration and can easily
+be extended to target new or custom infrastructure backends.
+
+Here we utilized these benefits to implement the Knative scale-from-zero logic in a special Mixer
+[out-of-process adapter](https://github.com/istio/istio/wiki/Mixer-Out-Of-Process-Adapter-Dev-Guide)
+which allows you to use any programming language and to build and maintain your extension as a stand-alone
+program without the need (as in previous Istio releases) to build the whole Istio proxy.
+The diagram below presents the Knative setup for scale-from-zero using our adapter.
+
+{{< image width="60%" link="knative-mixer-adapter.png" caption="Knative scale-from-zero" >}}
+
+In this setup, there is no need to change the routing from/to **Activator** for an idle application as in the original Knative setup.
+When the Istio proxy, inside an ingress gateway, receives a new request for an idle application, it informs **Mixer**, including all the
+relevant metadata information.
+**Mixer** then calls our adapter which triggers the Knative **Autoscaler** using the original Knative protocol.
+Note that we do not need to deal with buffering, retries and load-balancing because it is already handled by the Istio proxy.
```

Comment:
Instead of using the word Note, use a callout here:
https://preliminary.istio.io/about/contribute/creating-and-editing-pages/#callouts